govuk/sign-in/jwt/index.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+const path = require('path');
+const jwksClient = require('jwks-rsa');
+
+const createIssuerService = require('../issuer');
+
+const pKey = fs.readFileSync(path.resolve(process.cwd(), 'stuff/private_key.pem'));
+
+function createJwtService() {
+    function generateJWT(tokenContents, privateKey = pKey) {
+        return jwt.sign(tokenContents, privateKey, {algorithm: 'RS256'});
+    }
+
+    function decodeJWT(token) {
+        return jwt.decode(token);
+    }
+
+    // function getKey(header, callback) {
+    //     console.log({thing: this, header});
+    //     this.getSigningKey(header.kid, (err, key) => {
+    //         console.log({key});
+    //         if (err) {
+    //             throw err;
+    //         }
+    //         const signingKey = key.publicKey || key.rsaPublicKey;
+    //         callback(null, signingKey);
+    //     });
+    // }
+
+    // https://traveling-coderman.net/code/node-architecture/authentication/
+    async function verifyJWT(token) {
+        try {
+            const issuerService = createIssuerService();
+            const jwksUri = await issuerService.getJwtKeysUri();
+            const client = jwksClient({
+                jwksUri,
+                requestHeaders: {
+                    'user-agent': 'cica-web'
+                }
+            });
+
+            // eslint-disable-next-line no-inner-declarations
+            async function getKey(header, callback) {
+                await client.getSigningKey(header.kid, (err, key) => {
+                    if (err) {
+                        callback(err);
+                        return;
+                    }
+                    const signingKey = key.publicKey || key.rsaPublicKey;
+                    callback(null, signingKey);
+                });
+            }
+            jwt.verify(token, await getKey);
```

Comment:
This isn't a full verification. You need to verify the Audience, issuer and algorithms used as a minimum.